oss_src/unity/python/sframe/data_structures/sarray.py

```diff
@@ -1873,6 +1873,36 @@ def sample(self, fraction, seed=None):
         with cython_context():
             return SArray(_proxy=self.__proxy__.sample(fraction, seed))
 
+    def hash(self, seed=0):
+        """
+        Returns an SArray with a hash of each element. seed can be used
+        to change the hash function to allow this method to be used for
+        random number generation.
+
+        Parameters
+        ----------
+        seed : int
+            Defaults to 0. Can be changed to different values to get 
+            different hash results.
+
+        Returns
+        -------
+        out : SArray
+            An integer SArray with a hash value for each element. Identical
+            elements are hashed to the same value
+        """
+        with cython_context():
+            return SArray(_proxy=self.__proxy__.hash(seed))
+
+    @classmethod
+    def random_integers(self, size, seed=None):
+        """
+        Returns an SArray with random integer values.
+        """
+        if seed is None:
+            seed = time.time()
```

Comment:
At the moment, this is consistent with how sa.sample() works.I could fix it there. But then it will definitely be necessary to expand int to size_t. (or ssize_t). which will be a large change.